src/main/java/game_world/objects/events/ItemPickUpEvent.java

```diff
@@ -23,34 +25,61 @@ public class ItemPickUpEvent extends Event {
     private final ArrayList<String> inputs = new ArrayList<String>(Arrays.asList(
             "yes", "no"
     ));
+    public Inventory inventory;
+    public String text;
+    public ItemFactory itemFactory;
 
-    public ItemPickUpEvent(String name, String item) {
+    public ItemPickUpEvent(String name, String item, String text) {
         this.name = name;
         this.type = "Item Pick-Up";
         this.item = item;
         this.inputValidator = new WorldInputValidator(inputs);
         this.awaitInput = false;
         this.isDone = false;
+        this.inventory = Player.getInventory();
+        this.text = text;
     }
 
     @Override
     public void preInput() {
         this.awaitInput = true;
-        OutputHandler output = Output.getScreen();
-        // change text below
-        StringBuilder newMessage = new StringBuilder("[ITEM PICK-UP EVENT] What would you like to do?");
+        StringBuilder textToDisplay = new StringBuilder("[ITEM PICK-UP EVENT] " + this.text);
+        textToDisplay.append("\n Do you want to pick up the item?\n");
         for (String input : inputs) {
-            newMessage.append("\n\t◈ ").append(input);
+            textToDisplay.append("\t◈ ").append(input).append("\n");
         }
-        output.generateText(String.valueOf(newMessage));
+        Output.getScreen().generateText(textToDisplay.toString());
     }
 
     @Override
     public void postInput(String input) {
+        boolean success = false;
         this.awaitInput = false;
         this.isDone = true;
-        OutputHandler output = Output.getScreen();
-        output.generateText("You decided to " + input + ".");
+
+        if (input.equals("yes")){
+            switch (this.item) {
```

Comment:
what is the problem with just inventory.addItem(itemFactory.createItem(this.item))? why is this switch needed?